lib/server.js

```diff
@@ -271,6 +271,20 @@ function create(log, error, config, routes, db) {
     }
   )
 
+  const metricsContext = require('./metrics/context')(log, config)
+
+  server.decorate('request', 'stashMetricsContext', request => {
```

Comment:
> I don't like the way this looks...

Actually, it doesn't *have* to look like that, there is another way. Would you prefer the following?

```js
  server.decorate('request', 'stashMetricsContext', metricsContext.stash)
  server.decorate('request', 'gatherMetricsContext', metricsContext.gather)
  server.decorate('request', 'validateMetricsContext', metricsContext.validate)
```

I opted not to do it this way because it forces the `metricsContext` methods to use `this` rather than being passed a `request` argument. I thought that might confuse future readers of the code who weren't aware of these calls to `decorate`.

But I'm happy to do it like that if you prefer it. Yay or nay?